### PR TITLE
Miters

### PR DESCRIPTION
**Pull request by Przemog1**

### CODE REVIEW

File: include/nbl/builtin/hlsl/cpp_compat/intrinsics.h [hunk 1]
```diff
@@ -28,6 +27,10 @@ NBL_SIMPLE_GLM_PASSTHROUGH(lerp,mix)
 // transpose not listed cause it needs friendship
 
 #undef NBL_SIMPLE_GLM_PASSTHROUGH
+
+#include <algorithm>
```
Comment (Erfan-Ahmadi) [inline include/nbl/builtin/hlsl/cpp_compat/intrinsics.h:31]: چرا؟

Comment (Przemog1) [reply include/nbl/builtin/hlsl/cpp_compat/intrinsics.h:31]: looks like i forget to delete

File: include/nbl/builtin/hlsl/shapes/beziers.hlsl [hunk 2]
```diff
@@ -25,6 +27,12 @@ inline auto functionAlias(Args&&... args) -> decltype(origFunctionName(std::forw
 NBL_ALIAS_TEMPLATE_FUNCTION(std::min, min);
 NBL_ALIAS_TEMPLATE_FUNCTION(std::max, max);
 
+template<typename T>
+inline T rsqrt(T x)
+{
+    return 1.0f / std::sqrt(x);
+}
+
```
[the remark this answers is not in the document]
Comment (Przemog1) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:35]: its part of this todo
https://github.com/Devsh-Graphics-Programming/Nabla/blob/cad_shaders_work_miters/include/nbl/builtin/hlsl/shapes/beziers.hlsl#L12

[the remark this answers is not in the document]
Comment (Erfan-Ahmadi) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:35]: Hmmm did I put that to-do? Can't remember the context

[the remark this answers is not in the document]
Comment (Przemog1) [reply include/nbl/builtin/hlsl/shapes/beziers.hlsl:35]: i did, don't worry about that i'm already done with it

File: include/nbl/builtin/hlsl/limits.hlsl [hunk 3]
```diff
@@ -273,9 +273,9 @@ struct numeric_limits : std::numeric_limits<T>
     NBL_CONSTEXPR_STATIC_INLINE T epsilon = base::epsilon();
     NBL_CONSTEXPR_STATIC_INLINE T round_error = base::round_error();
 
-    NBL_CONSTEXPR_STATIC_INLINE uint_type quiet_NaN     = std::bit_cast(base::quiet_NaN());
-    NBL_CONSTEXPR_STATIC_INLINE uint_type signaling_NaN = std::bit_cast(base::signaling_NaN());
-    NBL_CONSTEXPR_STATIC_INLINE uint_type infinity      = std::bit_cast(base::infinity());
+    NBL_CONSTEXPR_STATIC_INLINE uint_type quiet_NaN     = std::bit_cast<uint_type, T>(base::quiet_NaN());
```
Comment (Erfan-Ahmadi) [inline include/nbl/builtin/hlsl/limits.hlsl:276]: I don't get this, Is this done by you or you merged some other branch?

Comment (Przemog1) [reply include/nbl/builtin/hlsl/limits.hlsl:276]: done by me, it wouldn't compile because of "cannot deduce T"

**Pull request merged**